File: Lua/utils/primality/iterativePrimalityTest.lua
-- iterativePrimalityTest.lua
-- 
-- 

local function test(number)
	if number <= 1 then return false end
	if number == 2 then return true end
	
	for i = 2, math.sqrt(number) do
		if number % i == 0 then 
			return false 
		end
	end
	
	return true
end

return test

File: CSharp/RandomNumberGeneration/BlumBlumShub/Utils/Primality/IterativePrimalityTest.cs
using System;

namespace RandomNumberGenerators.Utils.Primality {
    public class IterativePrimalityTest : IPrimalityTest {
        public bool Test(long number) {
            if(number <= 1) return false;
            if(number == 2) return true;

            for(int i = 2; i <= (int)Math.Ceiling(Math.Sqrt(number)); i++) {
                if(number % i == 0) return false;
            }

            return true;
        }
    }
}
